module/VuFind/src/VuFind/DigitalContent/OverdriveConnector.php

```diff
@@ -0,0 +1,1557 @@
+                    $this->debug("title locked in:");
+                } else {
+                    $result->msg
+                        = "Could not lock in Overdrive resourceID [$overDriveId]: "
+                        . $response->message;
+                }
+            } else {
+                //todo: translate
+                $result->code = 'od_code_connection_failed';
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * Get Configuration
+     * Sets up a local copy of configurations for convenience
+     *
+     * @return bool|\stdClass
+     */
+    public function getConfig()
+    {
+        $conf = new \stdClass();
+        if (!$this->recordConfig) {
+            $this->error(
+                "Could not locate the Overdrive Record Driver "
+                . "configuration."
+            );
+            return false;
+        }
+        if ($this->recordConfig->API->productionMode == false) {
+            $conf->productionMode = false;
+            $conf->discURL = $this->recordConfig->API->integrationDiscoveryURL;
+            $conf->circURL = $this->recordConfig->API->integrationCircURL;
+            $conf->libraryID = $this->recordConfig->API->integrationLibraryID;
+            $conf->websiteID = $this->recordConfig->API->integrationWebsiteID;
+        } else {
+            $conf->productionMode = true;
+            $conf->discURL = $this->recordConfig->API->productionDiscoveryURL;
+            $conf->circURL = $this->recordConfig->API->productionCircURL;
+            $conf->libraryID = $this->recordConfig->API->productionLibraryID;
+            $conf->websiteID = $this->recordConfig->API->productionWebsiteID;
+        }
+
+        $conf->clientKey = $this->recordConfig->API->clientKey;
+        $conf->clientSecret = $this->recordConfig->API->clientSecret;
+        $conf->tokenURL = $this->recordConfig->API->tokenURL;
+        $conf->patronTokenURL = $this->recordConfig->API->patronTokenURL;
+        $conf->idField = $this->recordConfig->Overdrive->overdriveIdMarcField;
+        $conf->idSubfield
+            = $this->recordConfig->Overdrive->overdriveIdMarcSubfield;
+        $conf->ILSname = $this->recordConfig->API->ILSname;
+        $conf->isMarc = $this->recordConfig->Overdrive->isMarc;
+        $conf->displayDateFormat = $this->mainConfig->Site->displayDateFormat;
+        $conf->consortiumSupport
+            = $this->recordConfig->Overdrive->consortiumSupport;
+        $conf->showMyContent
+            = strtolower($this->recordConfig->Overdrive->showMyContent);
+        $conf->noAccessString = $this->recordConfig->Overdrive->noAccessString;
+        $conf->tokenCacheLifetime
+            = $this->recordConfig->API->tokenCacheLifetime;
+        return $conf;
+    }
+
+    /**
+     * Returns an array of Overdrive Formats
+     *
+     * @return array
+     */
+    public function getFormatNames()
+    {
+        return [
+            'ebook-kindle' => "Kindle Book",
```

Comment:
These strings should probably be in a translation file and run through a translator somewhere or else they'll always display in English regardless of language setting.

Reply:
Moved to language file.